frontend/app/services/keyboard-shortcut-service.js

```diff
@@ -0,0 +1,153 @@
+//-- copyright
+// OpenProject is a project management system.
+// Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License version 3.
+//
+// OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+// Copyright (C) 2006-2013 Jean-Philippe Lang
+// Copyright (C) 2010-2013 the ChiliProject Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+//++
+
+module.exports = function($window, $rootScope, $timeout, PathHelper) {
+
+  // modalHelperInstance
+  // Mousetrap
+  // TODO: globals, should be wrapped in .constant
+
+  var accessKeys = {
+    preview: 1,
+    newWorkPackage: 2,
+    edit: 3,
+    quickSearch: 4,
+    projectSearch: 5,
+    help: 6,
+    moreMenu: 7,
+    details: 8
+  };
+
+  // maybe move it to a .constant
+  var shortcuts = {
+    '?': showHelpModal,
+    'up up down down left right left right b a enter': showHelpModal,
+    'g m': 'staticMyPagePath',
+    'g o': projectScoped('staticProjectPath'),
+    'g w p': projectScoped('staticProjectWorkPackagesPath'),
+    'g w i': projectScoped('staticProjectWikiPath'),
+    'g a': projectScoped('activityFromPath'),
+    'g c': projectScoped('staticProjectCalendarPath'),
+    'g n': projectScoped('staticProjectNewsPath'),
+    'g t': projectScoped('staticProjectTimelinesPath'),
+    'n w p': projectScoped('staticWorkPackageNewWithParametersPath'),
+
+    'g e': accessKey('edit'),
+    'g p': accessKey('preview'),
+    'd w p': accessKey('details'),
+    'm': accessKey('moreMenu'),
+    'p': accessKey('projectSearch'),
+    's': accessKey('quickSearch'),
+    'k': focusPrevItem,
+    'j': focusNextItem
+  };
```

Comment:
shouldn't there be something like

```javascript
 '8': accessKey('details')
```

in here? Otherwise, shortcut '8' does nothing for me.